rfd/0001-block-registry.md

```diff
@@ -0,0 +1,95 @@
+---
+authors: Grzegorz Uriasz (gorbak25@gmail.com)
+state: draft
+---
+
+# RFD 1 - Block registry
+
+## What
+
+Defines the way workspaces store their data by introducing a per agent block registry responsible for keeping track of the node storage while at the same time having the extensibility for future P2P operations.
+
+## Why
+
+To experiment with various storage solutions the workspace storage layer needs to be fully decoupled from the workspace runtime and a well defined interface between them should be used. Workspaces are currently defined as docker images so the storage layer should provide container like primitives while at the same time being compatible with VM's and providing enough flexibility for native Nix support.
```

Comment:
```suggestion
To experiment with various storage solutions the workspace storage layer needs to be fully decoupled from the workspace runtime and a well defined interface between them has to be used. Workspaces are currently defined as docker images so the storage layer should provide container like primitives while at the same time being compatible with VMs.
```

Reply:
I don't recall we decided to natively support Nix in the future.

Reply:
We didn't decide to support Nix natively but this was one of your talking points on the phone when discussing this RFD :)